lms/static/coffee/src/instructor_dashboard/membership.coffee

```diff
@@ -670,6 +670,564 @@ class AuthList
+  constructor:  (@email_lists, @$section, @$email_list_containers)  ->
+    @$query_endpoint = $(".email-lists-management").data('query-endpoint')
+    @$total_endpoint = $(".email-lists-management")
+      .data('total-endpoint')
+    @$temp_queries_endpoint = $(".email-lists-management")
+      .data('temp-queries-endpoint')
+    @$delete_saved_endpoint = $(".email-lists-management")
+      .data('delete-saved-endpoint')
+    @$delete_temp_endpoint = $(".email-lists-management")
+      .data('delete-temp-endpoint')
+    @$delete_bulk_temp_endpoint = $(".email-lists-management")
+      .data('delete-bulk-temp-endpoint')
+    for email_list in @email_lists
+      email_list.$container.addClass 'active'
+
+    @$get_est_btn = @$section.find("input[name='getest']'")
+    @$get_est_btn.click () =>
+      @reload_estimated()
+
+    @$startover_btn = @$section.find("input[name='startover']'")
+    @$startover_btn.click () =>
+      @delete_temporary()
+      $("#queryTableBody tr").remove()
+      @reload_estimated()
+
+    @$save_query_btn = @$section.find("input[name='savequery']'")
+    @$save_query_btn.click () =>
+      @send_save_query()
+
+    @$email_csv_btn = @$section.find("input[name='getcsv']'")
+    @$email_csv_btn.click () =>
+      sending_query = []
+      tab = $("#queryTableBody")
+      rows = tab.find("tr")
+      _.each rows, (row) ->
+        sending_query.push(row.getAttribute('query'))
+
+      send_data = sending_query.join(',')
+      url = @$email_csv_btn.data 'endpoint'
+      # handle csv special case
+      # redirect the document to the csv file.
+      url += '/csv'
+      url += "?rolename=instructor"
+      url += "&existing=" + encodeURIComponent(send_data)
+      location.href = url
+
+    @load_saved_queries()
+    @load_saved_temp_queries()
+
+    #poll for query status every 15 seconds
+    POLL_INTERVAL = 1000 * 15  # 15 seconds in ms
+    @poller = new window.InstructorDashboard.util.IntervalManager(
+      POLL_INTERVAL, => @load_saved_temp_queries()
+    )
+    @poller.start()
+
+    $('#addQuery').click =>
+      selected = @$email_list_containers.find('select.single-email-selector')
+        .children('option:selected')
+      #check to see if stuff has been filled out
+      if selected[1].text == "Section"
+        @arr = [{'text':selected[0].text, 'id':selected[0].id},
+                {'text':selected[1].text, 'id':selected[1].id},
+                {'text':selected[4].text, 'id':selected[4].id},
+                {'text':selected[5].text, 'id':selected[5].id}]
+        @arr_text = [selected[0].text, selected[1].text,
+                     selected[4].text, selected[5].text]
+
+      else
+        @arr = [{'text':selected[0].text, 'id':selected[0].id},
+                {'text':selected[1].text, 'id':selected[1].id},
+                {'text':selected[2].text, 'id':selected[2].id},
+                {'text':selected[3].text, 'id':selected[3].id}]
+        @arr_text = [selected[0].text, selected[1].text,
```

Comment:
I'm not seeing `@arr_text` referenced outside of this method; why are we assigning it as an instance variable instead of making it local?

The one other reference to `@arr_text` appears to be unrelated (the method
includes an assignment), so we probably definitely don't want this as an
instance member.